roiextractors/segmentationextractor.py

```diff
@@ -271,62 +205,26 @@ def get_movie_framesize(self):
         '''
         pass
 
-    @abstractmethod
-    def get_movie_location(self):
-        '''Raw file location on storage.
-
-        Returns
-        -------
-        raw_movie_file_location: str
-            location as a string
-        '''
-        pass
-
-    @staticmethod
-    def _pixel_mask_extractor(_raw_images_trans, _roi_idx):
-        '''An alternative data format for storage of image masks.
-
-        Returns
-        -------
-        pixel_mask: numpy array
-            Total pixels X 4 size. Col 1 and 2 are x and y location of the mask
-            pixel, Col 3 is the weight of that pixel, Col 4 is the ROI index.
-        '''
-        temp = np.empty((1, 4))
-        for i, roiid in enumerate(_roi_idx):
-            _np_raw_images_trans = np.array(_raw_images_trans[:, :, i])
-            _locs = np.where(_np_raw_images_trans > 0)
-            _pix_values = _np_raw_images_trans[_np_raw_images_trans > 0]
-            temp = np.append(temp, np.concatenate(
-                (_locs[0].reshape([1, np.size(_locs[0])]),
-                 _locs[1].reshape([1, np.size(_locs[1])]),
-                 _pix_values.reshape([1, np.size(_locs[1])]),
-                 roiid * np.ones([1, np.size(_locs[1])]))).T, axis=0)
-        return temp[1::, :]
-
-    @abstractmethod
-    def get_channel_names(self):
-        '''List of  channels in the recoding.
+    def get_sampling_frequency(self):
+        '''This function returns the sampling frequency in units of Hz.
 
         Returns
         -------
-        channel_names: list
-            List of strings of channel names
+        samp_freq: float
+            Sampling frequency of the recordings in Hz.
         '''
-        pass
+        return self._sampling_frequency
 
-    @abstractmethod
-    def get_num_channels(self):
-        '''Total number of active channels in the recording
+    def get_num_rois(self):
+        '''Returns total number of Regions of Interest in the acquired images.
 
         Returns
         -------
-        no_of_channels: int
-            integer count of number of channels
+        no_rois: int
+            integer number of ROIs extracted.
         '''
-        pass
+        return len(self.get_roi_ids())
 
-    @abstractmethod
     def get_images(self):
```

Comment:
@Saksham20 I don't think this should be abstract, but just raise a NotImplementedError if not overridden by a derived class